src/nodedef.cpp

```diff
@@ -1203,6 +1231,54 @@ void boxVectorUnion(const std::vector<aabb3f> &boxes, aabb3f *box_union)
 	}
 }
 
+/*!
+ * Helper function for fixed/leveled nodeboxes in getBoxUnionReturns.
+ * Returns the smallest box that contains all boxes
+ * in the vector. Box_union is expanded.
+ * @param[in]      features   used to decide whether the nodebox
+ *                            can be rotated
+ * @param[in, out] box_union  the union of the arguments
+ * @param          to_add     nodebox to add to union
+ * @param          nbt        node box type
+ */
+void rawUnionFixed(const ContentFeatures &features,
+		aabb3f *box_union,
+		const std::vector<aabb3f> &to_add, enum NodeBoxType nbt)
+{
+	// Raw union (fixed)
+	aabb3f half_processed(0, 0, 0, 0, 0, 0);
+	boxVectorUnion(to_add, &half_processed);
+	// Set leveled boxes to maximal
+	if (nbt == NODEBOX_LEVELED) {
+		// -0.5 + 127/64
+		half_processed.MaxEdge.Y = 1.484375f * BS;
+	} else if (nbt == NODEBOX_LEVELED_PLANTLIKE ||
+			nbt == NODEBOX_LEVELED_PLANTLIKE_ROOTED) {
+		half_processed.MaxEdge.Y = SAFE_SELECTION_BOX_LIMIT * BS;
+	}
+	if (features.param_type_2 == CPT2_FACEDIR ||
+			features.param_type_2 == CPT2_COLORED_FACEDIR) {
+		// Get maximal coordinate
+		f32 coords[] = {
```

Comment:
May I suggest: `f32 max = std::max({fabsf(half_processed.MinEdge.X), ...})` (`std::max` with an initializer list). That should allow you to get rid of the loop.